modules/ppcp-settings/src/SettingsModule.php

```diff
@@ -914,4 +916,21 @@ protected function is_gateway_enabled( string $gateway_name ): bool {
 
 		return $gateway_enabled === 'yes';
 	}
+
+	protected function unset_apm_payment_methods_when_not_eligible( ContainerInterface $container, array &$payment_methods ): void {
```

Comment:
I suggest to be explicit about the method dependencies, instead of passing the whole container, pass the concrete dependencies this method relies on. Also I suggest to not pass `$payment_methods` as reference but make the method return it, it would make it easier to reason about (it avoids the function mutating the caller state) and easier to unit test.